### ISSUE TITLE
`stop` event and `chainHead` responses

### ISSUE THREAD

**Issue by josepot** [paritytech/json-rpc-interface-spec#86]
While working on the tests for the [`@capi/substrate-client`](https://github.com/paritytech/capi/pull/65), I encountered an ambiguity that requires clarity.

Consider this scenario:
1. The client issues a request for `chainHead_unstable_header`.
2. Before the client receives a response, a `stop` event is triggered.

This raises some questions:

1. **Responsiveness**: Will the server still dispatch a response to the `chainHead_unstable_header` request, even after a `stop` event?
   
2. **Response Type**: If a response is sent, can we expect one of two outcomes?
   - The response is valid, implying the server fetched the required data before the `followSubscription` was invalidated.
   - If the `followSubscription` gets invalidated before data retrieval, will the response be a JSON-RPC error?

3. **Generalization**: Does this behavior also apply to operation requests like `chainHead_unstable_storage` et al?

To sum up, I am operating under the assumption that a response will always be sent. For operation requests, any response type would result in the consumer receiving an `ErrorDisjoined` rejection. For non-operation requests like `chainHead_unstable_header`, I plan to forward the response as it is.

**Comment by tomaka** [comment]
> If the followSubscription gets invalidated before data retrieval, the response will be a JSON-RPC error.

The response is actually `null`, not a JSON-RPC error. It's written in the spec.

There's also a note specifically for this:

> Note: As explained in the documentation of chainHead_unstable_follow, the JSON-RPC server reserves the right to kill an existing subscription and unpin all its blocks at any moment in case it is overloaded or incapable of following the chain. If that happens, chainHead_unstable_header will return null.

**Comment by josepot** [comment]
> > If the followSubscription gets invalidated before data retrieval, the response will be a JSON-RPC error.
> 
> The response is actually `null`, not a JSON-RPC error. It's written in the spec.

Thanks for pointing that out! I actually missed that, my bad!

However, what about the other requests: [`chainHead_unstable_call`](https://paritytech.github.io/json-rpc-interface-spec/api/chainHead_unstable_call.html#chainhead_unstable_call), [`chainHead_unstable_unpin`](https://paritytech.github.io/json-rpc-interface-spec/api/chainHead_unstable_unpin.html#chainhead_unstable_unpin), [`chainHead_unstable_storage`](https://paritytech.github.io/json-rpc-interface-spec/api/chainHead_unstable_storage.html#chainhead_unstable_storage), etc:

Will the server still dispatch a response, even after a stop event? will it be a JSON-RPC error? 🙏

**Comment by josepot** [comment]
Thanks! Ok, now I get it. Sorry about that. So, basically, the only call that may not receive a response if "caught in the middle" is `unpin`, which kinda makes sense.

So, it could happen that the client sends an `unpin` request before the client has received the stop event, however the server was about to send it (or it had already sent it, but the client hadn't received it yet), in which case the server won't bother to respond back to the `unpin` request... I mean, for the sake of consistency, I would much rather getting back a response with an error or something, but it's not a big deal, I guess. Thanks!
